common/http/src/limit.rs

```diff
@@ -112,43 +110,39 @@ impl<T> Drop for InnerSlot<T> {
 }
 
 struct Backoff {
-    step: Cell<u32>,
+    step: AtomicU32,
     base: u64,
     multipler: u64,
 }
 
 impl Backoff {
     pub fn new() -> Self {
         Self {
-            step: Cell::new(0),
+            step: AtomicU32::new(0),
             base: 2,
             multipler: 10,
         }
     }
 
-    pub fn snooze(&self) {
-        let step = self.step.get();
-        sleep(Duration::from_millis(self.base.pow(step) * self.multipler));
```

Comment:
Sleeping worker threads in an async runtime for ever increasing durations turns out to be a bad idea.